### PR TITLE
Simplify/Refactor buildutils.py

### PR DESCRIPTION
**Pull request by bryanwweber**
**Changes proposed in this pull request**

- Simplify/Refactor the `site_scons/buildutils.py` file. As mentioned in the [discussion post](https://github.com/Cantera/enhancements/discussions/93), this is going to be a long-running project. I'm planning to do the scripts/files in batches, so `buildutils.py` was the first.

**Checklist**

- [x] There is a clear use-case for this code change
- [x] The commit message has a short title & references relevant issues
- [x] Build passes (`scons build` & `scons test`) and unit tests address code coverage
- [x] The pull request is ready for review

### CODE REVIEW

**Comment by ischoegl** [comment]
:tada: ... as I mentioned elsewhere, this is not necessarily my strength. From what I can tell, this is largely to improve formatting, with a couple of more substantial changes. Big :+1: for `logger`, but `os.path` is very much alive. Snake case, quotes, etc. are as you mentioned elsewhere. Would you mind commenting on other parts you tried to tackle in this PR? Besides `build_utils.py` of course ...

**Comment by bryanwweber** [comment]
Thanks @ischoegl!

* My goal in general is to simplify the code, in part by using Python 3.6 and up features (f-strings and pathlib are the two that come immediately to mind). I'm also trying to update the code style to reflect modern Python practices and PEP-8. I guess there wasn't a whole lot of simplification that was able to be done to `buildutils.py`, but I think the changes here will make it easier to simplify other things later, or possibly add features. This is really the tip of the iceberg, so to speak.
* The changes in the `SCons*` files are almost entirely due to changing/removing functions in `buildutils.py`. There are a few places where there are unrelated changes to the `SCons*` files, but I think they're pretty small changes for now that I made incidentally while making other changes.
* I'm a big fan of `pathlib` and a big anti-fan of `os.path`. Operating purely on strings when it comes to paths is (can be) really tricky, especially when working between platforms. `pathlib` automatically handles things like different slash conventions, trailing slashes, etc. (Edit to add: the intention with this PR was not to replace `os.path` everywhere, yet. I'm going to go through the files in batches to simplify PR review).

File: site_scons/buildutils.py
Comment (bryanwweber) [inline site_scons/buildutils.py]: I need to add `"help"` here.

**Review by ischoegl — COMMENTED** [review]
No objections from my side - the CI appears to be happy, so I am happy (I don't have enough insights to go beyond).

**PS:** Regarding my comment about `os.path` - I noticed a couple of new imports for the 'un'preferred library, hence my comment. ... overall I second your preference for `pathlib`, but it appears the transition will come later, correct? From personal experience, replacing paths is laborious, and the diff is already quite busy because of the formatting changes. So leaving it for later is likely preferable.

**Comment by bryanwweber** [comment]
> the CI appears to be happy, so I am happy (I don't have enough insights to go beyond).

Thanks @ischoegl If there are any places where you see possible improvements in the Python, I'd be happy to know them!

> I noticed a couple of new imports for the 'un'preferred library, hence my comment. ... So leaving it for later is likely preferable.

Yes, that's my plan 😄 I started here by limiting the imports that `from buildutils import *` takes, which meant that all the stdlib imports in `buildutils` that were carried into other files were removed. And then did the minimum necessary in each of the `SCons*` files to get them working again.

I should also say that I'm trying to keep exact feature parity as I make these changes, including exactly the same output as before. So far, I've been successful 😃

File: site_scons/buildutils.py
Comment (ischoegl) [inline site_scons/buildutils.py:83]: docstring - I know it's inheriting from logging, but what is the objective?

File: site_scons/buildutils.py
Comment (ischoegl) [inline site_scons/buildutils.py:114]: docstring - I know it's inheriting from logging, but what is the objective?

File: site_scons/buildutils.py
Comment (ischoegl) [inline site_scons/buildutils.py]: here I'd almost recommend an exception to the line length, or to put things on their own line with `"""` (i.e. the old formatting was clearer)

File: site_scons/buildutils.py
Comment (ischoegl) [inline site_scons/buildutils.py]: does this fit on one line with 88 characters? This is one of the `black` formatting choices I don't understand.

**Review by ischoegl — COMMENTED** [review]
Alright :wink: ... I looked over `buildutils.py` and have a couple of minor points. Can give it a closer look later on.

Understood about `from xyz import *` ... linters really don't like it.

File: site_scons/buildutils.py
Comment (bryanwweber) [reply site_scons/buildutils.py]: I reformatted this, hopefully it's clearer now? The advantage is that the field width is a little more obvious now that I'm using `str.format`, the disadvantage is, why are you using `str.format` with string literal arguments? Ah well.

File: site_scons/buildutils.py
Comment (bryanwweber) [reply site_scons/buildutils.py]: Yes, this was this way from a previous point in the history when it was called `build_logger` rather than `logger`, and the `build_` tipped it over 88 characters. I'm not actually using Black to format the file 😄

File: site_scons/buildutils.py
Comment (ischoegl) [inline site_scons/buildutils.py]: You could invoke `sys.exit(TestState.FAIL)` here, which relates to the class below.

File: site_scons/buildutils.py
Comment (ischoegl) [inline site_scons/buildutils.py]: You start with PEP 484 above, but didn't continue ...

File: site_scons/buildutils.py
Comment (ischoegl) [inline site_scons/buildutils.py]: PEP 484 ... I believe `env` is specific to `scons`, but there are some ways to define custom types.

File: site_scons/buildutils.py
Comment (ischoegl) [inline site_scons/buildutils.py]: PEP 484. Docstring is not clear: what options are there?

File: site_scons/buildutils.py
Comment (ischoegl) [inline site_scons/buildutils.py]: `mglob` could use a better name; I recall that at some point it wasn't obvious that this was a cantera-specific command. And PEP 484, of course ...

File: site_scons/buildutils.py
Comment (ischoegl) [inline site_scons/buildutils.py]: Same, PEP 484 is used intermittently, but not here.

File: site_scons/buildutils.py
Comment (ischoegl) [inline site_scons/buildutils.py]: while this is internal, it is complex enough to use some comments

**Review by ischoegl — COMMENTED** [review]
While I believe the Python code overall looks good, some of the docstrings could use improvements. Also, you started applying PEP 484, but didn't follow through with all signatures. While I know that some don't like the appearance, I find those annotations tremendously helpful even when looking at code that I wrote myself (especially once short-term memory no longer kicks in).

File: site_scons/buildutils.py
Comment (ischoegl) [reply site_scons/buildutils.py]: Somewhat yes, although the original formatting wasn’t too bad either (only issue was the indent with the triple quotes). Your call ...

File: site_scons/buildutils.py
Comment (bryanwweber) [reply site_scons/buildutils.py]: Good point... I'm interested in your opinion about this. The reason I left them out of the functions that I did here is that they seemed more like noise and not so helpful for future use. It would require importing several libraries under the type hinting check, and I didn't think it would add enough value to include the types in the signature. That may change as I go through the other files, though. What do you think?

Comment (ischoegl) [reply site_scons/buildutils.py]: I *think* you can keep this relatively simple by using type aliases, i.e. `TypeVar` (see [here](https://www.python.org/dev/peps/pep-0484/#id17))
```
T = TypeVar('T', bound=...)
```
I.e. you just locally define an alias that documents what this is? I had to dig this up from something one of my students found at some point.

Comment (bryanwweber) [reply site_scons/buildutils.py]: I don't think it's necessary to use `TypeVar` here, a simple alias would be enough. I'll add types in a single commit, and let's see if it doesn't make the code illegible. If it looks terrible, I can just drop that commit.

File: site_scons/buildutils.py
Comment (bryanwweber) [reply site_scons/buildutils.py]: The intention of the `TestState` enum is to explicitly report the status of tests, not general exit status. I added a docstring to clarify.

File: site_scons/buildutils.py
Comment (bryanwweber) [reply site_scons/buildutils.py]: The options are defined in `SConstruct`. They are the build options, not options for this function.

File: site_scons/buildutils.py
Comment (bryanwweber) [reply site_scons/buildutils.py]: What comments do you think would help improve, beyond the top-level docstring?

File: site_scons/buildutils.py
Comment (bryanwweber) [reply site_scons/buildutils.py]: Alright, it didn't come out too bad. I didn't add types for unused arguments, but everything else should have one now.

File: site_scons/buildutils.py
Comment (ischoegl) [reply site_scons/buildutils.py]: that was clear :wink:, but PEP 484 by itself clarifies things here. (if this weren't an interface only seen by people digging deep under the hood, I would recommend to italicize *options* in the docstring)

File: site_scons/buildutils.py
Comment (ischoegl) [reply site_scons/buildutils.py]: It looks great :+1:, within the context that the aesthetics of PEP 484 may not be universally liked (I myself don't mind the annotations; on the contrary, I like the fact that it really helps when looking at an interface I am not familiar with).

Regarding unused arguments (e.g. 'target' and 'source' in `regression_test_message`), it may still help to know why they remain there (and what types they would be). You mentioned this in `print_report`, but it still is unclear what those required arguments represent.

File: site_scons/buildutils.py
Comment (ischoegl) [reply site_scons/buildutils.py]: The point is well taken, but it may not be universally known what the magic number `1` means for the exit status.

File: site_scons/buildutils.py
Comment (ischoegl) [reply site_scons/buildutils.py]: Looking over this again, I believe it's fine. Some of the context doesn't show up on when looking at the PR on the web interface (and trusty emacs isn't too helpful here either - although someone probably has written an extension). 

Regardless, opening the PR in VS code was a revelation ... :wink:

**Review by ischoegl — COMMENTED** [review]
@bryanwweber ... feel free to consider all of my comments as resolved. I'll leave a final approval to those with more intricate knowledge of the internal workings of `scons` (myself, I'd have to rely on GH Actions passing). Apart from that, I am pretty happy that I can make more sense of `buildutils.py` after the refactoring.

File: SConstruct
Comment (speth) [inline SConstruct]: While I generally agree that f-strings are great, I feel like the `quoted` function was easier to read, and there are multiple gotchas to writing this out the next time one needs the behavior of `quoted`, like needing to remember to use `!s`, and understanding how nesting of the different quote types works with the f-string braces.

File: SConstruct
Comment (speth) [inline SConstruct]: Are you trying to get rid of all usage of `print`? Why?

File: SConstruct
Comment (speth) [inline SConstruct:1478]: While you're renaming everything?
```suggestion
    install_root = Path.cwd().joinpath(env["stage_dir"], stage_prefix)
```

File: SConstruct
Comment (speth) [inline SConstruct:1630]: So are you replacing all of my beautiful single quotes with double quotes, or not? 😁

File: site_scons/buildutils.py
Comment (speth) [inline site_scons/buildutils.py]: I understand that need for some change because of `pathlib`, but I think the boolean expression is easier to read than the all/generator expression format when there are only two items.

File: site_scons/buildutils.py
Comment (speth) [inline site_scons/buildutils.py]: Maybe `TestResult` would be more clear? `TestState` makes me think about whether or not the test has run.

File: site_scons/buildutils.py
Comment (speth) [inline site_scons/buildutils.py]: The alias `TListify` is pretty inscrutable, and only used in this one place. Can you just write the actual set of viable types here?

**Review by speth — CHANGES_REQUESTED** [review]
I think most of my in-line comments are fairly minor.

The one slightly larger concern I have is that I don't really see the benefit to using the `logging` module here. I don't think it's possible to use any of this except when run as a script by SCons, so there's not really a case where this would need to interact with other logging activities, which is where the more complex logging machinery makes sense. This takes a chunk of additional setup in `build_utils.py`, plus has several end-use points where you are forced to do convoluted things like `logger.info(__doc__, extra={"print_level": False})` instead of the simple `print(__doc__)`.

File: SConstruct
Comment (bryanwweber) [reply SConstruct]: Yeah, I was thinking about this as well. Giving something a definite name (that is, `quoted`) has some value in indicating the intentionality of the change. I'll change it back.

File: SConstruct
Comment (bryanwweber) [reply SConstruct]: This is a great question, I'll address it in a top-level comment addressing the use of `logging` alltogether.

File: SConstruct
Comment (bryanwweber) [reply SConstruct:1478]: Renaming this here would require renaming it in other places in this file, which I was trying to avoid (to keep this PR focused on changing `buildutils.py` and the minimum set of changes to keep things working). A few things slipped through that don't strictly follow that, like adding the `logger` in a few places, but that was my general thought.

File: SConstruct
Comment (bryanwweber) [reply SConstruct:1630]: Same as the previous comment, this was a simple find-replace change for the `mglob` name. Don't you worry, those single quotes _must go_! 😄

**Comment by bryanwweber** [comment]
> The one slightly larger concern I have is that I don't really see the benefit to using the `logging` module here. I don't think it's possible to use any of this except when run as a script by SCons, so there's not really a case where this would need to interact with other logging activities, which is where the more complex logging machinery makes sense. This takes a chunk of additional setup in `build_utils.py`, plus has several end-use points where you are forced to do convoluted things like `logger.info(__doc__, extra={"print_level": False})` instead of the simple `print(__doc__)`.

> Are you trying to get rid of all usage of `print`? Why?

@speth I considered a few things in choosing the `logging` module here. I'm not totally sold that it's the right choice, so I'd be happy to discuss. My reasoning was:

1. The `logging` module allows us (as developers) to communicate intent by the method name instead of with the first few words of the message.
2. It makes it easier to be consistent with message formatting. `info` messages will always say `INFO` unless you explicitly disable that, likewise `warning`, `error`, and `debug`. Combined with the first reason above, this means no more forgetting to add context about how serious a problem you think this output is.
3. It should allow users to have finer control over the output. SCons is really noisy IMO and our configuration and various other output doesn't help that. It'd be nice to have a setting for "Yes, I'm pretty sure everything is going to work, just give me a progress bar" or even no output at all. I think it will be possible, with `logging`, for a user to set a variable in `cantera.conf` (or on the command line) and select the level of output, and I don't think that'd be possible (or at least much more difficult) just by using `print` calls.
4. It's more of a "set-it-and-forget-it" approach to output, in terms of where the output is sent. Right now, I've configured it so that `debug`, `info`, and `warning` output is sent to `stdout` and `error` and higher are sent to `stderr`. Using `print` calls would require setting the `file` kwarg for error messages if we want to send error output to `stderr` (which I believe would be an improvement).

Now, the counterarguments (which you've noted, and which I also thought of in working on this and balanced in favor of `logging`):

1. It's a lot of configuration.
2. One of the big advantages of `logging` is collecting messages from long-running processes when you need the program to keep running, but want a log of any output; or, interacting with/combining several libraries and combining their output. Obviously, as you note, SCons scripts are basically standalone, run-once scripts, so these advantages are moot.
3. The convoluted method of printing "regular" output instead of log messages, via the `print_level` Boolean.

So, those are the thoughts that I had. I did consider a couple of other options for terminal output as well... One was, if you look at the commit history, I'm pretty sure it's still there where I started with two `Logger` instances which avoided the problem of the weird Boolean hack. On the other hand, it wasn't clear when to use each logger, so I didn't like it ☹️ 

I similarly thought about using a `Logger` for INFO/WARNING/ERROR messages, and `print()` for "output", but again, it wasn't clear to me how I would explain to someone when to use which, so I didn't do it. I suppose there's a similar problem with "when to use `print_level=False`, but at least that's self-documenting.

Let me know what you think! Like I said, I'm not totally sold on using `logging` here, so I'd be happy to hear any further thoughts you have.

**Comment by ischoegl** [comment]
As a comment, I just checked out the branch to see what the new output would look like. On my Python 3.6.9 tool-chain (mostly vanilla Ubuntu 18.04), `scons` fails with the unhelpful error
```
docker@xps13:/src$ scons build
scons: Reading SConscript files ...
  File "/src/SConstruct", line 62

    logger.error(f"Unrecognized command line target: {command!r}")

                                                                ^

SyntaxError: invalid syntax

docker@xps13:/src$ python --version
Python 3.6.9
```

**PS:** I did run `scons clean` before (from `main`) ... and here's my `SCons` version

```
docker@xps13:/src$ scons --version
SCons by Steven Knight et al.:
	script: v3.0.1.74b2c53bc42290e911b334a6b44f187da698a668, 2017/11/14 13:16:53, by bdbaddog on hpmicrodog
	engine: v3.0.1.74b2c53bc42290e911b334a6b44f187da698a668, 2017/11/14 13:16:53, by bdbaddog on hpmicrodog
	engine path: ['/usr/lib/scons/SCons']
Copyright (c) 2001 - 2017 The SCons Foundation
```
which I believe is still the system's version (see toolchain configuration [here](https://github.com/ischoegl/cantera-docker/blob/master/ubuntu-toolchain/Dockerfile))

**Comment by bryanwweber** [comment]
Thanks @ischoegl can you try running python3 ``which scons`` Sometimes, especially for the system SCons, it pulls the system Python 2 because of the shebang in the script

**Comment by ischoegl** [comment]
I have `alias python=python3` set in `.bashrc` (which scons happily ignores, so I have another one in `cantera.conf`). Here's the output you requested:
```
docker@xps13:/src$ python3 which scons
python3: can't open file 'which': [Errno 2] No such file or directory
docker@xps13:/src$ which scons
/usr/bin/scons
```
The error for scons itself is somewhat surprising, as I've been using the toolchain without issues for a while ...

**Comment by bryanwweber** [comment]
@ischoegl  Sorry, that's a failure of the GitHub Markdown formatting. You need backticks around "which scons", so that the shell runs it as a command and inserts the output after the `python`. So (trying again): ``python3 `which scons` ``

I think the problem is with the Python that's running SCons being the system Python (that is, Python 2), not the Python used to build the interface (`python_cmd`). Unfortunately, these are distinct things for the system-installed SCons...

**Comment by ischoegl** [comment]
Alright, I was wondering! Here's the output ...
```
docker@xps13:/src$ python3 `which scons`
scons: Reading SConscript files ...

SCons build script for Cantera

Basic usage:
    'scons help' - print a description of user-specifiable options.

[...]
```

**Comment by bryanwweber** [comment]
Thanks! Yeah, so I need to find some way to interrupt SCons and produce a useful error message when it's using the system Python 2. I'm not sure how to do that, it seems that SCons tries to import all the SConstruct files first, and finds that syntax error... I'm not sure why the check I put into `buildutils.py` isn't being triggered.

**Comment by ischoegl** [comment]
Sorry, this may be a silly question, but this is a breaking change - how do I run this now? Imho, the expectation to type `scons build` as before and have things working isn't too far fetched. If it's the f-strings that cause this error, then they probably have to go.

File: .github/workflows/main.yml
Comment (ischoegl) [inline .github/workflows/main.yml:174]: I believe this has to remain

**Review by ischoegl — DISMISSED** [review]
I believe CI should check that this will continue to work. Ubuntu 18.04 LTS will be supported for another 2 years.

**Comment by speth** [comment]
> I'm not sure why the check I put into buildutils.py isn't being triggered.

The `SConstruct` file has to be *parsed* before it can get as far as trying to execute the `import` statement for `buildutils`. The best way of providing information about the problem that I can think of would be to stick an f-string near the top of `SConstruct` where the string itself contains the information telling the user what to do to fix it.

File: .github/workflows/main.yml
Comment (bryanwweber) [reply .github/workflows/main.yml:174]: Ubuntu 18.04 using Python 3 to run SCons is tested in the first build job. This only removes the case where Python 2 (that is, the system Python) is used to run SCons, which is no longer possible with this PR.

**Comment by bryanwweber** [comment]
> > I'm not sure why the check I put into buildutils.py isn't being triggered.
> 
> The `SConstruct` file has to be _parsed_ before it can get as far as trying to execute the `import` statement for `buildutils`. The best way of providing information about the problem that I can think of would be to stick an f-string near the top of `SConstruct` where the string itself contains the information telling the user what to do to fix it.

Alternatively, I could move that check from `buildutils.py` back into `SConstruct` at the top of the file, and raise something more useful than a `SyntaxError`, right?

**Comment by ischoegl** [comment]
@speth ... not necessarily elegant but likely very effective!

To document how this can be run anyways, it is (thanks @bryanwweber )
```
python3 `which scons` build
```

**Comment by bryanwweber** [comment]
@ischoegl @speth Here's the error message I came up with (from `buildutils.py`), let me know if it can be improved:

```python
if sys.version_info[:2] < (3, 6):
    logger.error(
        "Cantera must be built using Python 3.6 or higher. If you are using "
        "a system install of SCons, you can invoke SCons by writing\n"
        "    python3 `which scons`\n"
        "followed by any desired options."
    )
    sys.exit(1)
```

**Comment by ischoegl** [comment]
@bryanwweber ... error looks quite good but the system should know what the build option is so it *could* be more explicit?

**Comment by bryanwweber** [comment]
@ischoegl That same command must be used for any SCons things that one wants to do... `scons build`, `scons test`, `scons install`, etc. So I didn't want to be specific about any one option.

**Comment by speth** [comment]
> Alternatively, I could move that check from `buildutils.py` back into `SConstruct` at the top of the file, and raise something more useful than a `SyntaxError`, right?

No, the same thing applies. The whole file has to be parsed first before any statements are actually evaluated.

**Comment by ischoegl** [comment]
> The whole file has to be parsed first before any statements are actually evaluated.

Not using f-strings would resolve the syntax errors. There's another issue with `pathlib` (per discussion with @bryanwweber), but that's likely easier to resolve.

**Comment by ischoegl** [comment]
>  That same command must be used for any SCons things that one wants to do... scons build, scons test, scons install, etc. So I didn't want to be specific about any one option.

What I mean is that the build option added by the user should be known to SCons, and can potentially be inserted into the error message by constructing the string?

**Comment by bryanwweber** [comment]
> > That same command must be used for any SCons things that one wants to do... scons build, scons test, scons install, etc. So I didn't want to be specific about any one option.
> 
> What I mean is that the build option should be known, and can potentially be inserted into the error message by constructing the string?

Ah, I see. Based on what @speth said, I don't think that will be possible, since the options won't have been set up until the file is parsed (which is interrupted by the `SyntaxError`).

**Comment by ischoegl** [comment]
@bryanwweber ... I believe resolving the `SyntaxError` (by going back to older-style string formatting) would be the way to go, as everything else becomes much easier.

File: .github/workflows/main.yml
Comment (ischoegl) [reply .github/workflows/main.yml:174]: True. However, is there a way to check that the correct error is thrown in a GH action? I believe that's still worthwhile.

**Comment by bryanwweber** [comment]
> @bryanwweber ... I believe resolving the `SyntaxError` (by going back to older-style string formatting) would be the way to go, as everything else becomes much easier.

@ischoegl I've been thinking about this a lot over the weekend. I don't agree this is the case, because it then limits the entire file to be compatible syntactically with Python 2 and Python 3, which defeats some decent percentage of the point of this rewrite. For instance, we can't use `f`-strings, we can't use keyword-only arguments, we can't use modern-style type annotations, and probably more. Moreover, that caveat would only apply to `SConstruct`, and I know that I'd forget to only use compatible syntax in that one file.

So, after trying many alternatives, I've landed on deliberately throwing a `SyntaxError` that gives a useful message, as @speth suggested, as the most robust and the easiest to implement solution to this issue.

@speth Do you have any more comments/thoughts about using logging here, as a followup to [my comment](https://github.com/Cantera/cantera/pull/1012#issuecomment-825326643)?

**Comment by ischoegl** [comment]
@bryanwweber ... thanks for your response. It had been my assumption that the main issue with compatibility would be f-strings, hence my comment. I believe that there is a [work-around](https://stackoverflow.com/questions/35230635/type-hinting-in-python-2) for type hinting (PEP 484 is important to keep); regarding keywords-only, I'm not sure how much of a restriction it is, but I'm overall not sure that it's worth discussing further, as keeping Python 2 compatibility for the sake of having a better formatted error message is somewhat awkward.

Going forward, I believe there are two main things I hope this can accomplish:
* Make packaging/maintenance easier (which I believe is the point of this exercise)
* Ensure that things are reasonably robust in terms of older or deliberately stable systems (Ubuntu LTS, CentOS)

I'm definitely *not* arguing to leave Python 2 on life support ... . 

Regarding `logging`, I believe most of that information needs to be hidden (or redirected to a log file) by default, as showing some trivial logging info (e.g. `cti2yaml`) will only create questions on the UG even when there is no problem to start out with.

**Comment by bryanwweber** [comment]
> Regarding `logging`, I believe most of that information needs to be hidden (or redirected to a log file) by default, as showing some trivial logging info (e.g. `cti2yaml`) will only create questions on the UG even when there is no problem to start out with.

Thanks @ischoegl! Yes, I hope that for the `scons` scripts, adding `logging` will make redirecting/hiding output easier. I'm not planning any changes to the converter scripts, though, just the build scripts.

**Comment by ischoegl** [comment]
> I'm not planning any changes to the converter scripts, though, just the build scripts.

Agreed! However, `test_convert.py` appears to create a lot of logging output (at least on Ubuntu 18.04, *after* #1020 was merged) that I hope can be redirected one way or another.

**Comment by bryanwweber** [comment]
@speth in d9c4bf1 I think I found a better solution to the `print_level` weirdness. It can now be passed directly as a keyword argument to any of the logging functions. Let me know what you think.

**Comment by ischoegl** [comment]
Failing check is fixed by 0cc81add42976c4c3f654daae379ebf2d22285eb (per #1025) ... it should probably be applied to whatever is merged next as this part of CI is currently broken.

**Comment by bryanwweber** [comment]
> Failing check is fixed by [0cc81ad](https://github.com/Cantera/cantera/commit/0cc81add42976c4c3f654daae379ebf2d22285eb) (per #1025) ... it should probably be applied to whatever is merged next as this part of CI is currently broken.

Thanks @ischoegl My philosophy is that anyone who would run into that particular issue would be a pretty experienced developer and could either file an issue or see that the commit was in a pull request. Since that's a non-essential job, and #1025 won't take that much longer to get merged, I don't see a problem with leaving it on that PR.

File: site_scons/buildutils.py
Comment (speth) [inline site_scons/buildutils.py]: Given the `SyntaxError`-generating f-string in `SConstruct`, this is no longer necessary, right?

File: site_scons/buildutils.py
Comment (speth) [inline site_scons/buildutils.py]: I'm not convinced this is more readable than what was here.

**Review by speth — CHANGES_REQUESTED** [review]
Overall, I think I'm satisfied with the direction this has gone. There were just two minor suggested changes below.

I agree that there's some benefit to using `logging.info("...")` instead of `print("INFO: ...")`, and the `print_level` argument is not that bad for the handful of cases where we want to print something without a prefix.

Regarding a few of the comments, I'm not convinced that trying to reduce the verbosity of the build system would be productive. There are just too many cases where the details of the build log are the only thing that helps identify a problem. But this PR doesn't actually move in that direction, so this is kind of a moot point.

File: site_scons/buildutils.py
Comment (bryanwweber) [reply site_scons/buildutils.py]: Agreed!

**Review by speth — APPROVED** [review]

**Pull request merged**
